paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -585,70 +588,123 @@ def autoscale_marathon_instance(
-                'service_name': marathon_service_config.service,
-                'decision_policy': autoscaling_params[DECISION_POLICY_KEY],  # type: ignore
-                'paasta_cluster': marathon_service_config.cluster,
-                'instance_name': marathon_service_config.instance,
-            }
-            if yelp_meteorite:
-                gauge = yelp_meteorite.create_gauge('paasta.service.instances', meteorite_dims)
-                gauge.set(new_instance_count)
-                gauge = yelp_meteorite.create_gauge('paasta.service.max_instances', meteorite_dims)
-                gauge.set(marathon_service_config.get_max_instances())
-                gauge = yelp_meteorite.create_gauge('paasta.service.min_instances', meteorite_dims)
-                gauge.set(marathon_service_config.get_min_instances())
+            _record_autoscaling_decision(
+                marathon_service_config,
+                autoscaling_params,
+                utilization,
+                log_utilization_data,
+                error,
+                current_instances,
+                num_healthy_instances,
+                new_instance_count,
+                task_data_insufficient,
+                safe_downscaling_threshold,
+            )
     except LockHeldException:
         log.warning("Skipping autoscaling run for {service}.{instance} because the lock is held".format(
             service=marathon_service_config.service,
             instance=marathon_service_config.instance,
         ))
 
 
+def _record_autoscaling_decision(
+    marathon_service_config: MarathonServiceConfig,
+    autoscaling_params: AutoscalingParamsDict,
+    utilization: float,
+    log_utilization_data: Dict[str, str],
+    error: float,
+    current_instances: int,
+    num_healthy_instances: int,
+    new_instance_count: int,
+    task_data_insufficient: bool,
+    safe_downscaling_threshold: int,
+) -> None:
+    """
+    Based on the calculations made, perform observability side effects.
+    Log messages, generate time series, send any alerts, etc.
+    """
+    write_to_log(
+        config=marathon_service_config,
+        line=json.dumps(
+            dict(
+                timestamp=time.time(),
+                paasta_cluster=marathon_service_config.get_cluster(),
+                paasta_service=marathon_service_config.get_service(),
+                paasta_instance=marathon_service_config.get_instance(),
+                autoscaling_params=autoscaling_params,
+                utilization=utilization,
+                error=error,
+                current_instances=current_instances,
+                num_healthy_instances=num_healthy_instances,
+                new_instance_count=new_instance_count,
+                task_data_insufficient=task_data_insufficient,
+                safe_downscaling_threshold=safe_downscaling_threshold,
+            ),
+        ),
+        level='debug',
+    )
+    if new_instance_count != current_instances:
```

Comment:
I think duplicating this logic here and on line 601 is going to lead to these getting out of sync, which will mean log messages that say we do something different than we actually do.

Can we put these log lines back next to the actions they're describing, but keep the JSON log and the meteorite stuff?

Reply:
I hear what you're saying but it's super long so I pulled it over wholesale thinking it would be nice to keep all of the logging side effects together. Agree there is duplication on 601/602, I could try to drop that into a helper.

The way I was thinking of it is that the decision was already made in the helpers autoscale_marathon_instance calls, this is just grouping together some basic logic about the numbers to interpret what happened for a human (if people do read these?).

I actually feel a little bad about dumping json into what will also end up in the service's `paasta logs -C deploy`, not sure if you had a better pattern I could refer to for going straight to clog?

Reply:
I guess I'm wondering who those debug level messages were actually for? Now that we're printing the entire json, how does it sound to just drop them (new diff)?

Anyway it's a kind of transitionary state, I think that logging helper will still possibly run / initiate more math happening on the numbers to find, e.g., critical alerting levels. Not sure how you feel about this approach over vs an SLO oriented one, but wanted to get this out so we can talk more about it with some code to refer to.

Reply:
Yeah, if we think the new JSON line is at least as informative as the current debug messages (i.e. if we can deduce which code path would have been taken), I'm not tied to them. I like easily parsed logs.

In that interest, maybe add `task_data_insufficient`?

Reply:
face palm so hard, I actually removed the variable that initiated doing this whole thing.

Ok, new diff uploaded and fixed a type issue.